vpr/src/route/rr_graph_uxsdcxx_serializer.h

```diff
@@ -578,12 +578,17 @@ class RrGraphSerializer final : public uxsd::RrGraphBase<RrGraphContextTypes> {
                     inode, node.type());
             }
         } else {
-            node.set_side(from_uxsd_loc_side(side));
+            std::bitset<NUM_SIDES> side_mask = from_uxsd_loc_side(side);
+            for (const e_side& candidate_side : SIDES) {
+                if (side_mask[size_t(candidate_side)]) {
+                    node.set_side(candidate_side);
```

Comment:
This should be "node.add_side".  How did a unit test not catch this failure?  Add another test case?

Reply:
Good catch. I forgot to update this line. It is not detected because most rr_graph XML has only 1 side for rr_nodes.

Reply:
The second part of my comment was to suggest we specifically need a basic / strong test to check round tripping a design with more than 1 side on the rr_nodes, otherwise this code could easily go stale.

Reply:
I agree. As far as I know, the Titan architecture has multiple sides for many ``rr_nodes``. But its ``rr_graph.xml`` could be very large which may be an overkill in terms of complexity in basic/strong test. Do you have any small examples? I can add a new test.